nbextensions/bulkImportCell/bulkImportCell.js

```diff
@@ -377,42 +440,82 @@ define([
          * 1. if there's a tab showing, stop() it and detach it
          * 2. update the tabs state to be selected
          * @param {string} tab id of the tab to display
+         * @param {string} fileType id of the filetype we're swapping to
          */
-        function toggleTab(tab) {
+        function toggleTab(tab, fileType) {
+            // if we're toggling the currently selected tab off,
+            // then it should be turned off.
+            if (tab === state.tab.selected && tab !== null && !fileType) {
+                tab = null;
+            }
             state.tab.selected = tab;
+            stopWidget();
+
+            if (tab !== null) {
+                if (!fileType) {
+                    fileType = state.fileType.selected;
+                }
+                runTab(tab, fileType);
+            }
+            cellTabs.setState(state.tab);
+        }
+
+        function stopWidget() {
```

Comment:
Just decomposed this into its own function. This stops the currently selected widget, if there is one.